### PR TITLE
MULE-14171: SDK loses the Map DataType generics

### PR DESCRIPTION
**Pull request by estebanwasinger**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/message/Message.java [hunk 1]
```diff
@@ -160,13 +162,26 @@ static Message of(Object payload) {
...
     CollectionBuilder collectionValue(Object[] value);
 
+    /**
+     * Sets the map value for the {@link Message}'s payload to be built.
+     *
+     * @param value     the map content
+     * @param keyType   the map's key type
+     * @param valueType the map's value type
+     * @return this builder
+     * @throws NullPointerException if the content is null
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/message/Message.java]: `@since`

File: src/main/java/org/mule/runtime/api/message/Message.java [hunk 2]
```diff
@@ -242,37 +257,30 @@ static Message of(Object payload) {
      */
     CollectionBuilder itemMediaType(MediaType mediaType);
 
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    CollectionBuilder mediaType(MediaType mediaType);
```
Comment (afelisatti) [inline src/main/java/org/mule/runtime/api/message/Message.java]: Why are you deleting these?

Comment (estebanwasinger) [reply src/main/java/org/mule/runtime/api/message/Message.java]: they are interface overridings without sense, these are already defined by the `PayloadBuilder`, and mess it up the implementation. Probably Mule PR will be more clarifying

File: src/main/java/org/mule/runtime/api/message/Message.java (hunk 1 above)
Comment (estebanwasinger) [reply src/main/java/org/mule/runtime/api/message/Message.java]: Added `@since` in the interface declaration

**Pull request merged**